model/riscv_sys_control.sail

```diff
@@ -162,22 +174,24 @@ function check_CSR_access(csrrw, csrpr, p, isWrite) =
 function check_TVM_SATP(csr : csreg, p : Privilege) -> bool =
   not(csr == 0x180 & p == Supervisor & mstatus[TVM] == 0b1)
 
-function check_Counteren(csr : csreg, p : Privilege) -> bool =
-  match(csr, p) {
-    (0xC00, Supervisor) => mcounteren[CY] == 0b1,
-    (0xC01, Supervisor) => mcounteren[TM] == 0b1,
-    (0xC02, Supervisor) => mcounteren[IR] == 0b1,
+// There are several features that are controlled by machine/supervisor enable
+// bits (m/senvcfg, m/scounteren, etc.). This abstracts that logic.
+function mode_bit_enabled(p : Privilege, machine_bit : bit, supervisor_bit : bit) -> bool = match p {
+  Machine => true,
+  Supervisor => machine_bit == bitone,
+  User => machine_bit == bitone & (not(haveSupMode()) | supervisor_bit == bitone),
```

Comment:
Should there be a reference to the spec here that you need both S and M bits?

> The setting of a bit in mcounteren does not affect whether the corresponding bit in scounteren is writable. However, U-mode may only access a counter if the corresponding bits in scounteren and mcounteren are both set.

Also I was initially confused by the name name of the function. Maybe something like ` function feature_enabled_for_priv(p : Privilege, machine_enable_bit : bit, supervisor_enable_bit : bit) -> ` would be clearer?

Reply:
Changed name to your suggestion.

Reply:
Updated to your suggested name. I didn't add that bit of the spec because I'm going to use this for senvcfg/menvcfg too in a future PR.

I kind of think we should have a general mechanism for linking to the spec, but it doesn't look like they've got around to writing it in a linkable way yet (like [this](https://spec.ferrocene.dev/ownership-and-deconstruction.html) for example). :-/

Reply:
Sounds like ideally the privspec should publish versioned HTML versions to github pages which would allow linking easily.